src/app/common/common.tsx

```diff
@@ -784,14 +785,66 @@ function HeaderRenderer(props: any, hnum: number): any {
     return <div className={cn("title", "is-" + hnum)}>{props.children}</div>;
 }
 
-function CodeRenderer(props: any): any {
+function CodeRenderer(props:any): any {
     return <code className={cn({ inline: props.inline })}>{props.children}</code>;
 }
 
 @mobxReact.observer
-class Markdown extends React.Component<{ text: string; style?: any; extraClassName?: string }, {}> {
+class Markdown extends React.Component<{ text: string; style?: any; extraClassName?: string; codeSelect?: boolean}, {}> {
+    codeBlockList: Array<React.RefObject<HTMLElement>>    
+
+    constructor(props: any) {
+        super(props);  
+        this.codeBlockList = [];
+        if(this.props.codeSelect) {
+            let model = GlobalModel;
+            let inputModel = model.inputModel; 
+        }
+    }  
+
+    CodeBlockRenderer(props: any, codeSelect: boolean, ): any { 
+        let codeText = props.node.children[0].children[0].value;
+        if(codeText) {
+            codeText = codeText.replace(/\n$/, "") // remove trailing newline        
+        }
+        let blockRef: React.RefObject<HTMLPreElement> = React.createRef();
+        let clickHandler: (e: React.MouseEvent<HTMLElement>, blockIndex: number) => void;
+        let curBlockIndex: number;
+        if(codeSelect) {
+            let model = GlobalModel;
+            let inputModel = model.inputModel;
+            curBlockIndex = inputModel.addCodeBlockToCodeSelect(blockRef);
+            clickHandler = (e: React.MouseEvent<HTMLElement>, blockIndex: number) => {
+                inputModel.setCodeSelectSelectedCodeBlock(blockIndex);
+            };
+        } else {
+            clickHandler = (e: React.MouseEvent<HTMLElement>, blockIndex: number) => {
+                navigator.clipboard.writeText(codeText);
```

Comment:
I added a simple click to copy functionality for all code blocks in markdown. There is no ui currently. I think it might make sense to remove this for this pr, or I can put it in as a small nice to have. In the future we want proper "click to copy" button functionality.